In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

In [2]:
df = pd.read_csv('seeds.csv')
df.head()

area  perimeter  compactness  kernel_length  kernel_width  \
0  15.26      14.84       0.8710          5.763         3.312   
1  14.88      14.57       0.8811          5.554         3.333   
2  14.29      14.09       0.9050          5.291         3.337   
3  13.84      13.94       0.8955          5.324         3.379   
4  16.14      14.99       0.9034          5.658         3.562   

   asymmetry_coefficient  groove_length  species  
0                  2.221          5.220        0  
1                  1.018          4.956        0  
2                  2.699          4.825        0  
3                  2.259          4.805        0  
4                  1.355          5.175        0

In [3]:
X = df[['kernel_length','kernel_width','perimeter','asymmetry_coefficient']]
y = df[['groove_length']]

In [4]:
from sklearn.model_selection import train_test_split

X_train, X_test, y_train, y_test = train_test_split(X,y,test_size=0.25,random_state=42)

In [5]:
from sklearn.ensemble import RandomForestRegressor

regressor = RandomForestRegressor(n_estimators=200)

regressor.fit(X_train,y_train)

<ipython-input-5-7ccc628ad0ee>:3: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples,), for example using ravel().
  regressor.fit(X_train,y_train)


RandomForestRegressor(n_estimators=200)

In [6]:
y_pred = regressor.predict(X_test)

In [7]:
np.set_printoptions(precision=2)
print(y_pred)

[5.12 5.03 6.23 4.71 4.81 5.15 4.99 5.16 5.8  5.03 6.02 5.03 5.02 6.13
 5.12 6.05 4.78 4.88 5.15 5.86 5.84 5.29 5.13 6.07 6.42 5.08 5.68 5.97
 4.88 5.   4.94 5.28 5.82 5.17 6.02 5.63 5.03 6.01 6.07 5.21 5.1  5.19
 5.18 6.17 5.09 5.13 4.87 5.87 5.48 5.17 5.52 4.98 6.03]


In [8]:
from sklearn.metrics import r2_score

r2_score(y_pred,y_test)

0.8798320245694742

The r2_score function is commonly used as an evaluation metric for regression models, including linear regression.

It measures the proportion of the variance in the target variable that is predictable from the input features.

A higher R-squared score indicates a better fit of the regression model to the data, where 1 represents a perfect fit and 0 represents no relationship between the predicted and actual values.

An R-squared score of 0.8798 for the regressor indicates that approximately 87.98% of the variance in the target variable can be explained by the linear regression model's predictions.

This indicates a good fit of the model to the data.